less_users/tests/test_user_challenge.py

```diff
@@ -1,18 +1,119 @@
-from less_users.models import UserChallenge
+import pytest
 
+from freezegun import freeze_time
+from datetime import date
 
-def test_user_challenge_correct_creation(user, challenge_1_day, create_user_challenge):
-    user_challenge = create_user_challenge
+from less_users.models import UserChallenge, Log
+
+
+def test_user_challenge_correct_creation(
+    user, challenge_1_day, create_user_challenge_day
+):
+    user_challenge = create_user_challenge_day
 
     assert user_challenge in UserChallenge.objects.all()
     assert user_challenge.user == user
     assert user_challenge.challenge == challenge_1_day
 
 
-def test_user_challenge_deactivation(create_user_challenge):
-    user_challenge = create_user_challenge
-    user_challenge.is_active = False
+@pytest.mark.django_db
+@freeze_time("2023-01-01")
+def test_user_challenge_deactivation_when_duration_passed(user, challenge_1_day):
+    new_challenge = UserChallenge(
+        user=user,
+        challenge=challenge_1_day,
+    )
+    new_challenge.save()
+    new_challenge.days_left(todays_date=date(2023, 1, 5))
```

Comment:
po co to obliczasz jeżeli nigdzie nie sprawdzasz wartości?

Reply:
tu widzę, że nie działa jak bym tego chciała (szczególnie wpisywanie daty jako argument days_left)...